src/github/projectKit.ts

```diff
@@ -65,20 +101,39 @@ export class ProjectKit implements IProjectApi {
     private readonly logger: ILogger,
   ) {}
 
-  /* 
-  changeIssueStateInProject(issueCardId: number, state: "todo" | "in progress" | "blocked" | "done"): Promise<void> {
-    return this.gql(UPDATE_STATE_IN_PROJECT_QUERY, {
-      project: this.projectNumber,
-      item: issueCardId,
-      targetField: "Status",
-      targetFieldValue: state,
-    });
-  } */
+  async changeIssueStateInProject(issueCardId: string, project: NodeData, fields: FieldValues): Promise<void> {
+    try {
+      const op = await this.gql(UPDATE_PROJECT_V2_ITEM_FIELD_VALUE_QUERY, {
+        project: project.id,
+        item: issueCardId,
+        targetField: fields.field,
+        targetFieldValue: fields.value,
+      });
+
+      this.logger.debug("Returned " + JSON.stringify(op));
+    } catch (e) {
+      this.logger.error("Failed while executing the 'UPDATE_PROJECT_V2_ITEM_FIELD_VALUE_QUERY' query");
```

Comment:
I've got a rule of thumb that has never let me down: you should either handle the error (logging an error is handling), or throw it.  
Logging + throwing leads to errors being logged multiple times and messier debugging.  
Also, this log line just states that some error happened, without specifying which one. 
I suggest removing these try-catches completely.

Reply:
The reason for which I'm logging and throwing is because the error itself says something along the lines of "Could not find field id of object", but it doesn't say which query failed.

By doing this, when you check the logs, you'll see the error and which query failed.
Would you recommend a different approach? I need to keep the name of the query to identify the problem if one happens (or if GitHub changes their API).

Reply:
If you want to enrich error with some context info, I'd recommend wrapping one error object into another, and throwing newly created one